### ISSUE TITLE
template-injection: eliminate false positives with the `env.*` context

### ISSUE THREAD

**Issue by woodruffw** [woodruffw/zizmor#313]
Right now, there are a bunch of false positives when a `run:` context contains `${{ env.whatever }}`. In particular, the following is entirely safe, but we treat it as a potential injection:

```yaml
run: "echo ${{ env.foo }}"
env:
  foo: foo
```

We should do something similar to the static matrix check (but simplified, since there are thankfully no dimensions) for `env`. Specifically, we should **not** emit findings when the context is `env` **and** the referenced value is fully static (i.e. does not itself contain an expression).

**Comment by woodruffw** [comment]
See https://github.com/curl/curl/pull/15746 for some more context.

**Comment by woodruffw** [comment]
Thinking out loud: this isn't 100% trivial since each step/job/workflow `env` is a `LoE<Env>`, i.e. can be either a literal env mapping (with interior expressions) or a full-blown expression. So I think I'll start with just a `Step::env_key_is_static` API.
